src/trajectory.jl

```diff
@@ -686,7 +687,8 @@ function mh_accept_ratio(
     Horiginal::AbstractVector{<:T},
     Hproposal::AbstractVector{<:T},
 ) where {T<:AbstractFloat}
-    α = min.(1.0, exp.(Horiginal .- Hproposal))
+    α = min.(one(T), exp.(Horiginal .- Hproposal))
+    # FIXME: is this a bug?
     accept = _rand(rng) .< α
```

Comment:
similarly, this should generate a vector of IID uniform RVs.

Reply:
Only when `rng isa AbstractVector{<:AbstractRNG}}`.

Reply:
Right.

Reply:
It's actually this line (`rand(rng::AbstractRNG)` for `Horiginal ::AbstractVector{<: AbstractFloat}`) is buggy. Fixed now.